.github/scripts/prepare_JREs.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+
+URL_JDK8="https://api.adoptium.net/v3/binary/latest/8/ga/linux/x64/jre/hotspot/normal/eclipse"
+URL_JDK17="https://api.adoptium.net/v3/binary/latest/17/ga/linux/x64/jre/hotspot/normal/eclipse"
+
+mkdir -p JREs
+pushd JREs
+
+wget --content-disposition "$URL_JDK8"
+wget --content-disposition "$URL_JDK17"
+
+for file in *;
+do
+    mkdir temp
+
+    re='(OpenJDK([[:digit:]]+)U-jre_x64_linux_hotspot_([[:digit:]]+)(.*).tar.gz)'
+    if [[ $file =~ $re ]];
+    then
+        version_major=${BASH_REMATCH[2]}
+        version_trailing=${BASH_REMATCH[4]}
+
+        if [ $version_major = 17 ];
+        then
+            hyphen='-'
+        else
+            hyphen=''
+        fi
+        echo $hyphen
+
+        version_edit=$(echo $version_trailing | sed -e 's/_/+/g' | sed -e 's/b/-b/g')
```

Comment:
I'm not a big fan of piping from echo. You can do something like:

```
sed -e ... <<< `sed -e <<< ...`

```

To be fair that's still kinda cancer. In the end, it's up to personal preference, and the current code looks perfectly fine.

Reply:
Me either, but the pipes will be friendlier to any other dev who might work on it.